File: app/src/main/java/eu/qm/fiszki/ChoosenCategoryAdapter.java
package eu.qm.fiszki;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.CheckBox;
import android.widget.CompoundButton;

import java.util.ArrayList;

import eu.qm.fiszki.model.Category;
import eu.qm.fiszki.model.CategoryManagement;

/**
 * Created by bgood on 2016-04-03.
 */
public class ChoosenCategoryAdapter extends ArrayAdapter<Category> {

    ArrayList arrayList;
    LayoutInflater inflater;
    Context context;
    int rLayout;
    CheckBox checkBox;
    CategoryManagement categoryRepository;

    public ChoosenCategoryAdapter(Context context, int resource, ArrayList<Category> arrayList) {
        super(context, resource, arrayList);
        this.arrayList = arrayList;
        this.context = context;
        this.rLayout = resource;
        categoryRepository = new CategoryManagement(context);

    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final Category category = (Category) arrayList.get(position);
        LayoutInflater inflater = ((Activity) context).getLayoutInflater();
        convertView = inflater.inflate(rLayout, parent, false);
        checkBox = (CheckBox)convertView.findViewById(R.id.checkBox);
        checkBox.setText(category.getCategory());

        // here conditions

        return convertView;
    }

}

File: app/src/main/java/click/quickclicker/fiszki/settings/ChoosenCategoryAdapter.kt
package click.quickclicker.fiszki.settings

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.CheckBox
import click.quickclicker.fiszki.R
import click.quickclicker.fiszki.model.category.Category
import click.quickclicker.fiszki.model.category.CategoryRepository

class ChoosenCategoryAdapter(
    private val mContext: Context,
    private val rLayout: Int,
    private val arrayList: ArrayList<Category>
) : ArrayAdapter<Category>(mContext, rLayout, arrayList) {

    private val categoryRepository = CategoryRepository(mContext)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val category = arrayList[position]
        val inflater = (mContext as Activity).layoutInflater
        val view = inflater.inflate(rLayout, parent, false)
        val checkBox = view.findViewById<CheckBox>(R.id.checkBox)

        if (category.id == 1) {
            checkBox.setText(R.string.add_new_flashcard_no_category)
        } else {
            checkBox.text = category.getCategory()
        }

        checkBox.isChecked = category.isChosen

        checkBox.setOnCheckedChangeListener { _, isChecked ->
            category.isChosen = isChecked
            categoryRepository.updateCategory(category)
        }
        return view
    }
}
